kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h

```diff
@@ -0,0 +1,1234 @@
+            TSystemVectorType b_modified(total_size_of_system);
+
+            // Copy the RHS
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(BaseType::mEquationSystemSize); ++i) {
+                b_modified[i] = rb[i];
+            }
+            // Fill with zeros
+            #pragma omp parallel for
+            for (int i = static_cast<int>(BaseType::mEquationSystemSize); i < static_cast<int>(total_size_of_system); ++i) {
+                b_modified[i] = 0.0;
+            }
+            rb.resize(total_size_of_system, false);
+
+            // Compute LM contributions
+            TSystemVectorType b_lm(total_size_of_system);
+            ComputeRHSLMContributions(b_lm, mConstraintFactor);
+
+            // Fill auxiliar vector
+            TSparseSpace::UnaliasedAdd(b_modified, 1.0, b_lm);
+
+            // Finally reassign
+            TSparseSpace::Copy(b_modified, rb);
+
+            // Free memory
+            b_lm.resize(0, false);
+            b_modified.resize(0, false);
+        }
+
+        KRATOS_CATCH("")
+    }
+    
+    /**
+     * @brief Applies the constraints with master-slave relation matrix
+     * @param pScheme The integration scheme considered
+     * @param rModelPart The model part of the problem to solve
+     * @param rA The LHS matrix
+     * @param rb The RHS vector
+     */
+    void ApplyConstraints(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+
+        if (rModelPart.MasterSlaveConstraints().size() != 0) {
+            // Getting process info
+            const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+            
+            // First build the relation matrix
+            BuildMasterSlaveConstraints(rModelPart);
+
+            // Copy the LHS to avoid memory errors
+            TSystemMatrixType copy_of_A(rA);
+            TSystemMatrixType copy_of_T(BaseType::mT);
+            TSystemMatrixType transpose_of_T(TSparseSpace::Size2(BaseType::mT), TSparseSpace::Size1(BaseType::mT));
+            SparseMatrixMultiplicationUtility::TransposeMatrix<TSystemMatrixType, TSystemMatrixType>(transpose_of_T, BaseType::mT);
+
+            // Some common values
+            const SizeType number_of_slave_dofs = TSparseSpace::Size1(BaseType::mT);
+
+            // Definition of the total size of the system
+            const SizeType total_size_of_system = BaseType::mEquationSystemSize + (BaseType::mOptions.Is(DOUBLE_LAGRANGE_MULTIPLIER) ? 2 * number_of_slave_dofs : number_of_slave_dofs);
+            TSystemVectorType b_modified(total_size_of_system);
+
+            // Copy the RHS
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(BaseType::mEquationSystemSize); ++i) {
```

Comment:
why not using the `Space` function?

Reply:
True

Reply:
In fact if you check the sizes, this is not coping or setting the whole vector